app/src/main/java/com/github/swent/echo/viewmodels/HomeScreenViewModel.kt

```diff
@@ -0,0 +1,243 @@
+package com.github.swent.echo.viewmodels
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.swent.echo.authentication.AuthenticationService
+import com.github.swent.echo.compose.components.searchmenu.FiltersContainer
+import com.github.swent.echo.compose.components.searchmenu.SortBy
+import com.github.swent.echo.data.SAMPLE_EVENTS
+import com.github.swent.echo.data.SAMPLE_TAGS
+import com.github.swent.echo.data.model.Event
+import com.github.swent.echo.data.model.Tag
+import com.github.swent.echo.data.repository.Repository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import java.time.ZonedDateTime
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+enum class Overlay {
+    NONE,
+    EVENT_INFO_SHEET,
+    SEARCH_SHEET
+}
+
+enum class MapOrListMode {
+    MAP,
+    LIST
+}
+
+@HiltViewModel
+class HomeScreenViewModel
+@Inject
+constructor(
+    private val repository: Repository,
+    private val authenticationService: AuthenticationService,
+) : ViewModel() {
+
+    private val _overlay = MutableStateFlow<Overlay>(Overlay.NONE)
+    val overlay = _overlay.asStateFlow()
+    private val _mode = MutableStateFlow<MapOrListMode>(MapOrListMode.MAP)
+    val mode = _mode.asStateFlow()
+    private var filterTagSet: Set<Tag> = setOf()
+    private lateinit var allEventsList: List<Event>
+    private lateinit var allTagSet: Set<Tag>
+    private val _displayEventList = MutableStateFlow<List<Event>>(listOf())
+    val displayEventList = _displayEventList.asStateFlow()
+    private val _displayEventInfo = MutableStateFlow<Event?>(null)
+    val displayEventInfo = _displayEventInfo.asStateFlow()
+    private val _filtersContainer =
+        MutableStateFlow(
+            FiltersContainer(
+                searchEntry = "",
+                epflChecked = true,
+                sectionChecked = true,
+                classChecked = true,
+                pendingChecked = true,
+                confirmedChecked = true,
+                fullChecked = true,
+                from = ZonedDateTime.now(),
+                to = ZonedDateTime.now().plusDays(365),
+                sortBy = SortBy.NONE
+            )
+        )
+    val filtersContainer = _filtersContainer.asStateFlow()
+
+    private val _profileName =
+        MutableStateFlow<String>("John Doe") // placeholder until we have user profiles
+    val profileName = _profileName.asStateFlow()
+    private val _profileClass =
+        MutableStateFlow<String>("IN - BA6") // placeholder until we have user profiles
+    val profileClass = _profileClass.asStateFlow()
```

Comment:
`profileName ` and `profileClass` are not yet used. It would be cool if the hamburger menu would use those values.

Reply:
oh yeah I forgot I added those x) I'll change that